.github/workflows/integration_tests.yml

```diff
@@ -49,189 +49,171 @@ env:
-        elif [[ "${{ github.event.label.name }}" == "${{ env.triggerLabelQuick }}" ]]; then
-          echo "::set-output name=requested_tests::auto"
-        else
-          # For manual PR testing, use auto mode.
-          echo "::set-output name=requested_tests::auto"
+        if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then 
+          if [[ "${{ github.event.inputs.test_pull_request }}" != "sdk" ]]; then
+            # Triggered manually
+            echo "::set-output name=trigger::manual_trigger"
+            if [[ "${{ github.event.inputs.use_expanded_matrix }}" == "1" ]]; then
+              echo "::set-output name=requested_tests::expanded"
+            fi
+            if [[ -z "${{github.event.inputs.test_pull_request}}" ]]; then
+              # test_pull_request not specified
+              echo "::set-output name=github_ref::$GITHUB_SHA"
+            elif [[ "${{github.event.inputs.test_pull_request}}" == *:* ]]; then
+              # If specified as pr:commit_hash, split them.
+              echo "::set-output name=github_ref::$(echo ${{ github.event.inputs.test_pull_request }} | cut -d: -f2)"
+              echo "::set-output name=pr_number::$(echo ${{ github.event.inputs.test_pull_request }} | cut -d: -f1)"
+            else
+              # Just the PR specified, use refs/pull/<number>/merge as the ref.
+              echo "::set-output name=github_ref::refs/pull/${{github.event.inputs.test_pull_request}}/merge"
+              echo "::set-output name=pr_number::${{ github.event.inputs.test_pull_request }}"
+            fi
+          elif [[ "${{ github.event.inputs.test_pull_request }}" == "sdk" ]]; then
+            # Triggered by scheduled packaging SDK workflow.
+            echo "::set-output name=trigger::scheduled_trigger"
+            echo "::set-output name=github_ref::$GITHUB_SHA"
+            echo "::set-output name=requested_tests::expanded"
+          fi
+        elif [[ "${{ github.event_name }}" == "schedule" ]]; then
+          echo "::set-output name=trigger::scheduled_trigger"
+          echo "::set-output name=github_ref::$GITHUB_SHA"
+          echo "::set-output name=requested_tests::expanded"
+        elif [[ "${{ github.event_name }}" == "pull_request" ]]; then
+          echo "::set-output name=github_ref::$GITHUB_SHA"
+          if [[ "${{ github.event.action }}" == "labeled" && "${{ github.event.label.name }}" == "${{ env.triggerLabelPrefix }}"* ]]; then
+            echo "::set-output name=trigger::label_trigger"
+            echo "::set-output name=pr_number::${{ github.event.pull_request.number }}"
+            if [[ "${{ github.event.label.name }}" == "${{ env.triggerLabelQuick }}" ]]; then
+              echo "::set-output name=requested_tests::auto"
+            else
+              echo "::set-output name=requested_tests::expanded"
+            fi
+          elif [[ "${{ github.event.action }}" == "closed" && "${{ github.event.pull_request.merged }}" == "true" ]]; then
```

Comment:
Just a nit, isn't it slightly safer to say:
("${{ github.event.action }}" == "closed") && "${{ github.event.pull_request.merged }}"

(in case github.event.pull_request.merged happens to be "1" instead of "true" or some other non-zero value.